test/utils/application.data.spec.ts

```diff
@@ -163,6 +173,67 @@ describe("Application data utils", () => {
     expect(response).toEqual(false);
   });
 
+  test('findBeneficialOwner should return BO Individual if valid data given', () => {
+    const beneficialOwner = findBeneficialOwner(
+      APPLICATION_DATA_UPDATE_BO_MOCK,
+      PARAM_BENEFICIAL_OWNER_INDIVIDUAL,
+      BO_IND_ID
+    );
+    expect(beneficialOwner.id).toEqual(BO_IND_ID);
+  });
+
+  test('findBeneficialOwner should return BO Gov if valid data given', () => {
+    const beneficialOwner = findBeneficialOwner(
+      APPLICATION_DATA_UPDATE_BO_MOCK,
+      PARAM_BENEFICIAL_OWNER_GOV,
+      BO_GOV_ID
+    );
+    expect(beneficialOwner.id).toEqual(BO_GOV_ID);
+  });
+
+  test('findBeneficialOwner should return BO Other if valid data given', () => {
+    const beneficialOwner = findBeneficialOwner(
+      APPLICATION_DATA_UPDATE_BO_MOCK,
+      PARAM_BENEFICIAL_OWNER_OTHER,
+      BO_OTHER_ID
+    );
+    expect(beneficialOwner.id).toEqual(BO_OTHER_ID);
+  });
+
+  test('findBeneficialOwner should return undefined if invalid id given', () => {
+    const beneficialOwner = findBeneficialOwner(
+      APPLICATION_DATA_UPDATE_BO_MOCK,
+      PARAM_BENEFICIAL_OWNER_OTHER,
+      'fake-id'
+    );
+    expect(beneficialOwner).toEqual(undefined);
+  });
+
+  test('findBeneficialOwner should return undefined if invalid benficialOwnerType given', () => {
+    const beneficialOwner = findBeneficialOwner(
+      APPLICATION_DATA_UPDATE_BO_MOCK,
+      'not-a-real-bo-type',
+      BO_OTHER_ID
+    );
+    expect(beneficialOwner).toEqual(undefined);
+  });
+
+  test('checkGivenBoDetailsExist returns true if beneficial owner found', () => {
+    expect(checkGivenBoDetailsExist(
+      APPLICATION_DATA_UPDATE_BO_MOCK,
+      PARAM_BENEFICIAL_OWNER_GOV,
+      BO_GOV_ID
+    )).toEqual(true);
+  });
+
+  test('checkGivenBoDetailsExist returns true if beneficial owner found', () => {
```

Comment:
Test name needs to be changed here - is the same name as previous test

Reply:
good spot, changed